### ISSUE TITLE
Tabs are not handled correctly when errors are rendered

### ISSUE THREAD

**Issue by aartamonau** [mrkkrp/megaparsec#524]
The issue is similar to https://github.com/mrkkrp/megaparsec/issues/239. While the example in that issue works fine, the one below still produces incorrect results:

```
$ stack ghci --package megaparsec-9.3.0 --resolver lts-20.21
ghci> import Text.Megaparsec
ghci> import Text.Megaparsec.Char
ghci> import Data.Void
ghci> parseTest (string' "test" *> char '\t' *> char 'a' :: Parsec Void String Char) "test\tb"
1:9:
  |
1 | test        b
  |         ^
unexpected 'b'
expecting 'a'
```

**Comment by aartamonau** [comment]
I think the issue might be in this line: https://github.com/mrkkrp/megaparsec/blob/3496aaa34f66224eb2977f89b641bdac4ad507e4/Text/Megaparsec/Stream.hs#L622.

It aligns the position to the nearest column divisible by tab width (or something like that). This code predates `expandTab` which just replaces all tabs with tab width spaces here: https://github.com/mrkkrp/megaparsec/blob/3496aaa34f66224eb2977f89b641bdac4ad507e4/Text/Megaparsec/Stream.hs#L580.

So if I apply this diff:

```diff
                       id
                 | ch == tabTok ->
                     St
-                      (SourcePos n l (mkPos $ c' + w - ((c' - 1) `rem` w)))
+                      (SourcePos n l (mkPos $ c' + w))
                       (g . (fromTok ch :))
                 | otherwise ->
                     St
```

then the example in my original post works:

```
ghci> parseTest (string' "test" *> char '\t' *> char 'a' :: Parsec Void String Char) "test\tb"
1:13:
  |
1 | test        b
  |             ^
unexpected 'b'
expecting 'a'
```

**Comment by mrkkrp** [comment]
This is certainly a bug, although I think the position should indeed be `1:9`, so the current calculation of `SourcePos` is correct. The problem is that `expandTab` should be more intelligent and insert only as many spaces as is necessary to reach the next tab stop.

**Comment by mrkkrp** [comment]
@aartamonau Thanks for spotting it! I've just published version 9.3.1 with the fix.
